transaction.go

```diff
@@ -35,56 +35,68 @@ const (
 	TransactionSourceMerchant       TransactionSource = "merchant"
 )
 
+type TransactionPaymentInstrumentType string
```

Comment:
I think this could be named `PaymentInstrumentType`. That's what the enum is called in the Braintree .NET SDK.

https://github.com/braintree/braintree_dotnet/blob/882a6ea/src/Braintree/Transaction.cs#L110-L130